### PR TITLE
Refactored hook handlers in mod_pubsub

### PR DESCRIPTION
**Pull request by pawlooss1**
This PR changes all hook handlers in `mod_pubsub` module to `gen_hook` format.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/154735) / small_tests / d7b5b80ec8562808223dcec764e4ddcf8a0891a1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3879/154735/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3879/154735/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-29_09.34.53/lib.mongooseim.logs/run.2022-11-29_09.34.54/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/154820) / small_tests / 929be7574852929e4206fd59119c6562e8cf340e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3879/154820/small_tests.24.3.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3879/154820/small_tests.24.3.4/small/ct_run.mongooseim@localhost.2022-11-29_12.28.53/lib.mongooseim.logs/run.2022-11-29_12.28.54/suite.log.html)

File: src/mod_ping.erl [hunk 1]
```diff
@@ -151,7 +151,7 @@ iq_ping(Acc, _From, _To, #iq{sub_el = SubEl} = IQ, _) ->
 
 -spec handle_remote_hook(Acc, Params, Extra) -> {ok, Acc} when
     Acc :: term(),
-    Params :: #{tag := atom(), args := term(), c2s_state := ejabberd_c2s:state()},
+    Params :: #{tag := atom(), hook_args := term(), c2s_state := ejabberd_c2s:state()},
```
Comment (NelsonVides) [inline src/mod_ping.erl:154]: How comes dialyzer didn't complain here 😢

**Review by NelsonVides — APPROVED** [review]
LGTM 👌🏽

**Pull request merged**